src/mpl/include/mpl_thread_qthreads.h

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+/* This file is used when configured with (MPICH_THREAD_PACKAGE_NAME ==
+ * MPICH_THREAD_PACKAGE_QTHREADS) */
+
+#ifndef MPL_THREAD_QTHREADS_H_INCLUDED
+#define MPL_THREAD_QTHREADS_H_INCLUDED
+
+#include "mpl.h"
+#include "qthread.h"
+#include "qthread/tls.h"
+
+#include <errno.h>
+#include <assert.h>
+
+typedef qthread_spinlock_t MPL_thread_mutex_t;
```

Comment:
@janciesko is `qthread_spinlock_t` only available in Qthreads 1.19 and up? I had trouble getting that tarball to install on our Jenkins nodes. 1.18 installed fine, but is met with compilation errors.

Reply:
Yes, it requires version `1.19` or later. `1.19` added that type in order to support what MPICH required for Qthreads support.

Reply:
@raffenet Feel free to post the error that occurs when using 1.19 in the Jenkins job. It theory, it should work comparably to 1.18.

Reply:
It was actually errors when trying to build 1.19 from the release tarball. Here is the output.
```
System Characteristics:
       Target Style:
         Multi-node: no
       Topology API: libnumaV2
        Qtimer type: clock_gettime
     Aligned_t size: 8 (aligned on 8 byte boundaries)
 Default Stack size: 16kB

Safety/Debugging:
   Sanity assert()s: no
    Check alignment: no
          Profiling: none
   Debugging Output: no
        Guard Pages: no
          Precaching: no

Speed:
          Scheduler: sherwood
         Sinc Style: donecount
        Alloc Style: base
      Barrier Style: feb
   Dictionary Style: shavit
    Lazy Thread IDs: yes
       Pools/caches: memory
Increments/CAS/FEBs: Compiler Builtin (both), lock-based hash

Miscellany:
      Eureka Events: no

[raffenet@pmrs-centos64-240-04]/var/tmp/raffenet/qthreads-1.19% make -j8 install
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/sh '/var/tmp/raffenet/qthreads-1.19/config/missing' aclocal-1.16 -I config
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/sh '/var/tmp/raffenet/qthreads-1.19/config/missing' autoconf
 cd . && /bin/sh /var/tmp/raffenet/qthreads-1.19/config/missing automake-1.16 --foreign
/bin/sh ./config.status --recheck
running CONFIG_SHELL=/bin/sh /bin/sh ./configure --prefix=/nfs/gce/projects/pmrs/opt/qthreads-1.19 --no-create --no-recursion
./configure: line 4: .: filename argument required
.: usage: . filename [arguments]
make: *** [config.status] Error 2
make: *** Waiting for unfinished jobs....
[raffenet@pmrs-centos64-240-04]/var/tmp/raffenet/qthreads-1.19%
```
Running `autogen.sh` seems to have cleared up the issue locally, so I'll install that and give it a shot.

Reply:
Reproduced. Working on solution.

Reply:
Resolved. @raffenet, the tarball has been updated.